optimade/server/entry_collections/entry_collections.py

```diff
@@ -43,22 +96,114 @@ def get_attribute_fields(self) -> set:
                 attributes = attributes[next_key]
         return set(attributes["properties"].keys())
 
-    @abstractmethod
-    def find(
-        self, params: EntryListingQueryParams
-    ) -> Tuple[List[EntryResource], int, bool, set]:
+    def handle_query_params(
+        self, params: Union[EntryListingQueryParams, SingleEntryQueryParams]
+    ) -> dict:
+        """ Parse and interpret the backend agnostic query parameter models into a dictionary
+        that can be used by the specific backend. Currently returns the pymongo
+        interpretation of the parameters, which will need modification for modified
+        for other backends.
+
+        Parameters:
+            params: the initialized query parameter model from the server.
```

Comment:
```suggestion
            params (Union[EntryListingQueryParams, SingleEntryQueryParams]): The initialized query parameter model from the server.
```

Reply:
Is mkdocs not smart enough to use the typehint here?

Reply:
I've seen it happen for some things, but not all. If you find that it works, then no reason to put extra bytes in here 👍